Remove abi extractor

LuisLara-UH opened the pull request:
- [X] Sem Tests Passed

LuisLara-UH committed:
Add output decoder to solidity ABI.

Pending:
Transform output from Call or Invoke command.
Testing.

swapnilraj committed:
Convert buffer to string and add small fixes and TODOs

LuisLara-UH committed:
Merge branch 'develop' into transcode-warp-output

LuisLara-UH committed:
Update decoding

LuisLara-UH committed:
Add decoding to callOrInvoke command

LuisLara-UH committed:
Update encoding.

LuisLara-UH committed:
Remove divmod duplicated function

swapnilraj committed:
Declare cleanup

LuisLara-UH committed:
Small updates

LuisLara-UH committed:
Fix regex for getting felts width

LuisLara-UH committed:
Use FunctionFragment for parsing params

Update encoding

LuisLara-UH committed:
Merge branch 'develop' into transcode-warp-output

LuisLara-UH committed:
Convert decoded outputs to string

LuisLara-UH committed:
Small updates

LuisLara-UH committed:
Remove functionSignatureParser
Update decode outputs

LuisLara-UH committed:
remove abi extractor

LuisLara-UH committed:
Update cli.test

LuisLara-UH committed:
Merge branch 'transcode-warp-output' into remove-abi-extractor

swapnilraj commented on docs/docs/features/cairo_stubs.mdx:
```diff
@@ -102,7 +102,7 @@ namespace Example:
 end
 
-# Original soldity abi: ["constructor()","recursiveAdd(uint8,uint8)"]
+# Original solidity abi: [["constructor()", ""],["recursiveAdd(uint8,uint8)", "uint8"]]
```
We can delete this

swapnilraj commented on docs/docs/getting_started/a-usage-and-installation.mdx:
```diff
@@ -234,7 +234,7 @@ namespace foo:
 end
 
-# Original soldity abi: ["constructor(uint256)",""]
+# Original solidity abi: [["constructor(uint256)",""]]
```
this as well

swapnilraj commented on src/ast/ast.ts:
```diff
@@ -51,9 +51,8 @@ import { CairoFunctionDefinition } from './cairoNodes';
-  // SourceUnit id -> function signatures
-  abi: Map<number, Set<string>> = new Map();
+  // SourceUnit id -> function [signatures, returns]
+  abi: Map<number, Set<[string, string]>> = new Map();
```
if the abi extractor is deleted there should be no writes into this; so we can delete this

swapnilraj commented on src/io.ts:
```diff
@@ -5,7 +5,7 @@ import { TranspileFailedError, logError } from './utils/errors';
 import { AST } from './ast/ast';
 
-export const solABIPrefix = '// Original soldity abi:';
+export const solABIPrefix = '// Original solidity abi:';
```
delete

swapnilraj commented on src/starknetCli.ts:
```diff
@@ -198,7 +213,7 @@ export async function runStarknetCallOrInvoke(
   try {
     [funcName, inputs] = await encodeInputs(
-      filePath,
+      solAbiPath ?? '',
```
we should try to regenerat the solidity abi if its not found;
we have the path to solidity file

LuisLara-UH replied:
The solidity abi file is always generated unless it fails to compile, in which case it returns immediately and it gets not here. The same happens with abiPath, which is being used in the command later. That undefined checking is pure formality with JS. However, if you want I can put the code for generating the path there

swapnilraj commented on src/utils/utils.ts:
```diff
@@ -569,3 +569,7 @@ export function getContainingSourceUnit(node: ASTNode): SourceUnit {
   return root;
 }
+
+export function dumpABI(node: SourceUnit, ast: AST): string {
```
we don't need this since the abi mapping will empty now

swapnilraj commented on tests/cli.test.ts:
```diff
@@ -48,7 +49,7 @@ const tests: encodeTest[] = [
   tests.map(([signature, input, output]) =>
     it(`parses ${signature}`, () => {
-      const result = parse(signature)(input);
+      const result = encode(FunctionFragment.fromString(signature).inputs, input);
```
nice!

LuisLara-UH committed:
Fix encode tests

LuisLara-UH committed:
Merge branch 'transcode-warp-output' into remove-abi-extractor

LuisLara-UH committed:
Update requested changes

LuisLara-UH committed:
Remove some references

swapnilraj commented on src/starknetCli.ts:
```diff
@@ -210,18 +225,26 @@ export async function runStarknetCallOrInvoke(
+      const decodedOutputs = await decodeOutputs(
+        solAbiPath ?? '',
+        options.function,
+        warpOutput.toString().split(' '),
```
Now that we have the components of a struct in the abi we can improve the formatting and put the key names for a struct and for arrays we should wrap them in []

swapnilraj commented on tests/cli.test.ts:
```diff
@@ -33,7 +34,7 @@ const tests: encodeTest[] = [
   ['mint(bool,bool,bool,bool)', ['true', 'false', '1', '0'], ['0x1', '0x0', '0x1', '0x0']],
   ['mint(bytes8)', ['0xffffffffffffffff'], ['0xffffffffffffffff']],
-  ['byte_me(bytes)', [['0xff', '0xff', '0xff', '0xff']], ['0x4', '0xff', '0xff', '0xff', '0xff']],
+  ['byte_me(bytes[])', [['0xff', '0xff', '0xff', '0xff']], ['0x4', '0xff', '0xff', '0xff', '0xff']],
```
are the cli tests run in CI?

LuisLara-UH replied:
Yup, I realized of the encoding bugs because they were failing in CI

LuisLara-UH committed:
Merge branch 'develop' into remove-abi-extractor

LuisLara-UH committed:
Update export.ts

GuyMcComb commented:
Sem Tests Failing:
tests/behaviour/solidity/test/libsolidity/semanticTests/array/copying/array_copy_different_packing

LuisLara-UH commented:
`tests/behaviour/solidity/test/libsolidity/semanticTests/array/copying/array_copy_different_packing`

This semantic test is failing in develop too: the problem is that there is a contract named `c`, which is also the name of a variable inside. It is easily patched just changing the name of the contract. I think it is not an issue from this branch, and we could fix it on another PR.

LuisLara-UH committed:
Add formatting to decoded outputs

LuisLara-UH commented:
I added the new formatting for the decoded outputs, where I added the special formatting for structs and arrays. I tested it with a few cases, but I think we can implement some testing for decoding, like the one made for encoding (https://github.com/NethermindEth/warp/blob/develop/tests/cli.test.ts#L9)

swapnilraj commented:
> `tests/behaviour/solidity/test/libsolidity/semanticTests/array/copying/array_copy_different_packing`
> 
> This semantic test is failing in develop too: the problem is that there is a contract named `c`, which is also the name of a variable inside. It is easily patched just changing the name of the contract. I think it is not an issue from this branch, and we could fix it on another PR.

That's the wrong fix, we can't change the contract to make it work for our tests. Can you share the test result?
@GuyMcComb

GuyMcComb committed:
Merge branch 'develop' into remove-abi-extractor
